source/thead_rvv/fp32/prelu.c

```diff
@@ -53,28 +54,59 @@ int shl_rvv_prelu_fp32(struct csinn_tensor *input, struct csinn_tensor *alpha,
             output->layout = CSINN_LAYOUT_NC1HWC0;
         }
     } else if (input->layout == CSINN_LAYOUT_NCHW) {
-        for (int n = 0; n < input->dim[0]; ++n) {
-            for (int c = 0; c < input->dim[1]; ++c) {
-                float a = alpha_data[c];
-                int inner_size = input->dim[2] * input->dim[3];
-                while (inner_size > 0) {
-                    int vl = vsetvl_e32m2(inner_size);
-                    vfloat32m2_t _input = vle32_v_f32m2(input_data, vl);
-                    vbool16_t _mask = vmflt_vf_f32m2_b16(_input, 0.0f, vl);
-                    vfloat32m2_t _res = vfmul_vf_f32m2_m(_mask, _input, _input, a, vl);
-                    vse32_v_f32m2(output_data, _res, vl);
-                    input_data += vl;
-                    output_data += vl;
-                    inner_size -= vl;
+        if (alpha->dim[1] == csinn_tensor_size(alpha)) {
+            // simplify the calculation by avoiding broadcast
+            for (int n = 0; n < input->dim[0]; ++n) {
+                for (int c = 0; c < input->dim[1]; ++c) {
+                    float a = alpha_data[c];
+                    int inner_size = input->dim[2] * input->dim[3];
+                    while (inner_size > 0) {
+                        int vl = vsetvl_e32m2(inner_size);
+                        vfloat32m2_t _input = vle32_v_f32m2(input_data, vl);
+                        vbool16_t _mask = vmflt_vf_f32m2_b16(_input, 0.0f, vl);
+                        vfloat32m2_t _res = vfmul_vf_f32m2_m(_mask, _input, _input, a, vl);
+                        vse32_v_f32m2(output_data, _res, vl);
+                        input_data += vl;
+                        output_data += vl;
+                        inner_size -= vl;
+                    }
                 }
             }
-        }
-        if (output->layout == CSINN_LAYOUT_NC1HWC0) {
-            const int packn = csrr_vlenb() / sizeof(float);
-            output->dim[1] *= packn;
-            output->dim[4] = 0;
-            output->dim_count = 4;
-            output->layout = CSINN_LAYOUT_NCHW;
+        } else {
+            // broadcast alpha
+            int new_alpha_size = csinn_tensor_size(input);
+            float *alpha_data_b = shl_mem_alloc(new_alpha_size * sizeof(float));
```

Comment:
Minor:

```suggestion
            int input_size = csinn_tensor_size(input);
            float *alpha_data_b = shl_mem_alloc(input_size * sizeof(float));
```

Then you can just use `input_size` instead of calculation `ele_size` on L86